profiler/src/ProfilerEngine/Datadog.Profiler.Native/ManagedThreadList.cpp

```diff
@@ -193,6 +211,22 @@ uint32_t ManagedThreadList::Count()
     return static_cast<uint32_t>(_threads.size());
 }
 
+uint32_t ManagedThreadList::GetHighCountAndReset()
+{
+    std::lock_guard<std::recursive_mutex> lock(_mutex);
+    auto currentHigh = _highCount;
+    _highCount = static_cast<uint32_t>(_threads.size());
+    return currentHigh;
+}
+
+uint32_t ManagedThreadList::GetLowCountAndReset()
+{
+    std::lock_guard<std::recursive_mutex> lock(_mutex);
+    auto currentLow = _lowCount;
+    _lowCount = static_cast<uint32_t>(_threads.size());
+    return (currentLow == 0) ? _lowCount : currentLow;
+}
+
```

Comment:
Why not using `std::atomic` + `exchange` method. We won't need any locking here.

Reply:
Because we need to access safely to _threads.size that is protected by the same lock during thread creation/destruction